org.eclipse.transformer/src/main/java/org/eclipse/transformer/Transformer.java

```diff
@@ -1010,17 +1023,20 @@ public boolean setOutput() {
 
 		} else {
 			int indexOfLastSlash = inputName.lastIndexOf('/');
-			if (indexOfLastSlash == -1) {
-				useOutputName = OUTPUT_PREFIX + inputName;
-			} else {
-				String inputPrefix = inputName.substring(0, indexOfLastSlash + 1);
-				String inputSuffix = inputName.substring(indexOfLastSlash + 1);
-				useOutputName = inputPrefix + OUTPUT_PREFIX + inputSuffix;
-			}
+			File parent = inputFile.getParentFile();
+			File output = new File(parent, OUTPUT_PREFIX + inputFile.getName());
+			useOutputName = options.normalize(output.getAbsolutePath());
 		}
 
 		File useOutputFile = new File(useOutputName);
-		String useOutputPath = useOutputFile.getAbsolutePath();
+		String useOutputPath = null;
+		try {
+			useOutputFile = useOutputFile.getCanonicalFile();
+			useOutputPath = useOutputFile.getCanonicalPath();
+		} catch (IOException e) {
+			getLogger().error(consoleMarker, "Output error [ {} ] [ {} ]", outputName, e.toString());
```

Comment:
```suggestion
			getLogger().error(consoleMarker, "Output error [ {} ] [ {} ]", outputName, e.toString(), e);
```

exception as final arg